### PR TITLE
Adding a quick dirty feature to stop the bot after the time specified…

### PR DESCRIPTION
**Pull request by ngkogkos**
Short Description: 

Using time(), the bot can now exit if user specifies time in minutes.

### CODE REVIEW

**Comment by surfaace** [comment]
missing configuratin in config.json.example

**Comment by ngkogkos** [comment]
Cheers for that surfaace!

**Comment by surfaace** [comment]
is it tested on windows btw?

**Comment by ngkogkos** [comment]
You are total right. SIGALRM wont work on Windows. Will try to work around it.

File: pokecli.py [hunk 1]
```diff
@@ -268,6 +261,10 @@ def main():
 
             while True:
                 bot.take_step()
+                if config.stop_after and time.time() > stop_after:
+                    logger.log('Time is up, will exit now..', 'yellow')
+                    # TODO: write a function for bot metrics and call it here too
```
Comment (cwild) [inline pokecli.py]: raise SystemExit

File: pokecli.py [hunk 2]
```diff
@@ -268,6 +261,10 @@ def main():
...
+                    sys.exit(0)
 
         except KeyboardInterrupt:
```
Comment (cwild) [inline pokecli.py]: except (KeyboardInterrupt, SystemExit):

**Comment by cwild** [comment]
I just added some in-line comments to the last commit. I realise it's pretty dirty to raise a SystemExit but think it could be the quickest way without refactoring the metrics stuff out of the KeyboardInterrupt handler..

**Comment by ngkogkos** [comment]
@cwild I am fine with that. I am not a dev myself, so my moto is: if it works, it works. However, I think the metrics display code would anyway need refactoring one day in the future. I can just update the PR again to reflect your suggestions later on.

**Comment by BarryReid** [comment]
Any time line on getting this merged in?

**Comment by ngkogkos** [comment]
I am just waiting for opinions on the changes that @cwild suggested. I am fine with any way, just need the timer :D

**Comment by Zgrkaralar** [comment]
👍

**Comment by maestroi** [comment]
Would be nice if there is more of an anti ban system.
something like:

1. Random logout wait random minuts to log bag in and play more
2. Take breaks at like cafés to stay there.
3. sometimes wait at pokestops with lures to camp there

that would be more human like

**Comment by BarryReid** [comment]
Why has this been closed and not merged into dev?

**Comment by douglascamata** [comment]
Don't remember, lol. Is the dev interested in reworking for the current code's state?

**Comment by ngkogkos** [comment]
I can't really do it now, maybe later on.

**Pull request closed without merging**